.github/workflows/license-checker.yml

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+# This workflow checks that the BUSL license is not mentioned anywhere in
+# a PR targeting a release that should maintain the MPL-2.0 license.
+name: License Checker
+
+on:
+  pull_request:
+    types: [opened]
+    branches:
+      - release/1.14.*
+      - release/1.15.*
+      - release/1.16.*
```

Comment:
I added versions I expect people to be backporting into these days. We can go back further if people think it's warranted.
I also wanted to make sure not to run on `release/1.17.x` once it exists since code in that release will be under BUSL.